common/reflect_resource.go

```diff
@@ -702,6 +719,22 @@ func StructToData(result any, s map[string]*schema.Schema, d *schema.ResourceDat
 			}
 			log.Printf("[TRACE] set %s %#v", fieldPath, nv)
 			return d.Set(fieldPath, nv)
+		case schema.TypeBool, schema.TypeInt:
+			fv := v.FieldByName("ForceSendFields")
+			forceSendFields := []string{}
+			// force send fields might not exist in struct, so checking it against zero value first
+			if fv != (reflect.Value{}) {
+				forceSendFields = fv.Interface().([]string)
+			}
+
+			// If the field is not in the force send fields, we should not store it in the state
+			// Only bool and int values are in force send fields now
+			if len(forceSendFields) > 0 && !slices.Contains(forceSendFields, valueField.sf.Name) {
+				log.Printf("[TRACE] skipping field %s", valueField.sf.Name)
+				return nil
+			}
```

Comment:
Any chance we can define a single function to remove duplication between this and collectionToMaps?